### PR TITLE
Add models for annotation endpoints

### PR DESCRIPTION
**Pull request by zhx828**

### CODE REVIEW

File: core/src/main/java/org/mskcc/cbio/oncokb/apiModels/annotation/AnnotateStructuralVariantQuery.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import org.mskcc.cbio.oncokb.model.StructuralVariantType;
+
+/**
+ * Created by Hongxin Zhang on 2019-03-25.
+ */
+public class AnnotateStructuralVariantQuery extends AnnotationQuery {
+    private GeneQueryPair geneA;
+    private GeneQueryPair geneB;
+    private StructuralVariantType structuralVariantType;
+    private Boolean isFusion = false;
```
Comment (jjgao) [inline core/src/main/java/org/mskcc/cbio/oncokb/apiModels/annotation/AnnotateStructuralVariantQuery.java]: maybe change to `isFunctionalFusion`

File: core/src/main/java/org/mskcc/cbio/oncokb/model/CopyNumberAlterationType.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package org.mskcc.cbio.oncokb.model;
+
+/**
+ * Created by Hongxin Zhang on 2019-03-25.
+ */
+public enum CopyNumberAlterationType {
+    AMPLIFICATION,
+    DELETION,
```
[the remark this answers is not in the document]
Comment (zhx828) [reply core/src/main/java/org/mskcc/cbio/oncokb/model/CopyNumberAlterationType.java]: We do support it already

File: core/src/main/java/org/mskcc/cbio/oncokb/apiModels/annotation/AnnotateMutationByProteinChangeQuery.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package org.mskcc.cbio.oncokb.apiModels.annotation;
+
+import org.mskcc.cbio.oncokb.model.AlterationType;
+
+/**
+ * Created by Hongxin Zhang on 2019-03-25.
+ */
+public class AnnotateMutationByProteinChangeQuery extends AnnotationQuery{
+    private GeneQueryPair gene;
+    private String alteration;
```
Comment (zhx828) [inline core/src/main/java/org/mskcc/cbio/oncokb/apiModels/annotation/AnnotateMutationByProteinChangeQuery.java]: @jjgao should this be proteinChange?

File: core/src/main/java/org/mskcc/cbio/oncokb/apiModels/annotation/AnnotateMutationByHGVSg.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package org.mskcc.cbio.oncokb.apiModels.annotation;
+
+/**
+ * Created by Hongxin Zhang on 2019-03-25.
+ */
+public class AnnotateMutationByHGVSg extends AnnotationQuery{
+    private GeneQueryPair gene;
```
Comment (zhx828) [inline core/src/main/java/org/mskcc/cbio/oncokb/apiModels/annotation/AnnotateMutationByHGVSg.java]: Going to remove the gee, HGVSg does not require that.

**Pull request closed without merging**